app/process_proposal.go

```diff
@@ -35,6 +40,21 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) abci.ResponsePr
 		}
 	}
 
+	squareSize := data.SquareSize
+
+	// validate proposed square size
+	if !isValidSquareSize(int(squareSize), int(minSquareSize), int(maxSquareSize)) {
```

Comment:
[question] given this line assumes `isValidSquareSize` returns true for a valid square, why didn't the unit test that was added in this PR identify the logic error inside `isValidSquareSize`?